lib/models/device-type.ts

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2016 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import type { InjectedDependenciesParam, PineOptions } from '..';
+import { DeviceType } from '../types/models';
+import { mergePineOptions } from '../util';
+import * as errors from 'balena-errors';
+
+const getDeviceTypeModel = function (deps: InjectedDependenciesParam) {
+	const { pine } = deps;
+
+	const exports = {
+		/**
+		 * @summary Get all deviceTypes
+		 * @name getAll
+		 * @public
+		 * @function
+		 * @memberof balena.models.deviceTypeType
+		 *
+		 * @param {Object} [options={}] - extra pine options to use
+		 * @fulfil {Object[]} - device types
+		 * @returns {Promise}
+		 *
+		 * @description
+		 * This method returns all supported device types.
+		 *
+		 * @example
+		 * balena.models.deviceTypeType.getAll().then(function(deviceTypes) {
+		 * 	console.log(deviceTypes);
+		 * });
+		 *
+		 * @example
+		 * balena.models.deviceType.getAll({ $select: ['name', 'slug'] }).then(function(deviceTypes) {
+		 * 	console.log(deviceTypes);
+		 * })
+		 *
+		 * @example
+		 * balena.models.deviceType.getAll(function(error, deviceTypes) {
+		 * 	if (error) throw error;
+		 * 	console.log(deviceTypes);
+		 * });
+		 */
+		async getAll(options?: PineOptions<DeviceType>): Promise<DeviceType[]> {
+			if (options == null) {
+				options = {};
+			}
+
+			const deviceType = await pine.get({
+				resource: 'device_type',
+				options: mergePineOptions({ $orderby: 'name asc' }, options),
+			});
+			return deviceType;
+		},
```

Comment:
TODO: Add `async getAllSupported(options?: PineOptions<DeviceType>): Promise<DeviceType[]>`
which on top of `getAll` also does
```
                $filter: {
                    is_default_for__application: {
                        $any: {
                            $alias: 'idfa',
                            $expr: { idfa: {
                                is_host: true,
                                is_archived: false,
                            } },
                        },
                    }
                },
```